views.py

```diff
@@ -105,27 +88,35 @@ def post(self):
 class ReadPost(Resource):
     @api.response(404, 'Post not found!')
     def get(self, post_id):
-        """
-        Interact with a specific post
-        """
+        """ Interact with a specific post. """
         # Get specific post using the post_id
         post = Posts.query.filter_by(id=post_id).first()
         if not post:
             return {'message': 'Post not found!'}, 404
         else:
+            post = Posts.query.filter_by(id=post_id).first()
             post_schema = PostSchema()
             output = post_schema.dump(post).data
-            return jsonify({'post': output})
+            # Check if there's an image file
+            if post.image_file != None:
+                # Get the id
+                img_id = post.image_file
+                # Search for the img in the post image files
+                img_url = glob.glob(os.path.join(POST_UPLOAD_PATH, '{}.*'.format(img_id)))
+                # Attach it and jsonify the output
+                output['image_url'] = img_url[0]
+                return jsonify({'post': output})
+            else:
+                return jsonify({'post': output})
```

Comment:
we can eliminate this `else` by putting the `return` outside the `if`

Reply:
Alright